source/common/router/rds_impl.cc

```diff
@@ -226,15 +230,23 @@ RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getStaticRouteConfi
 }
 
 ProtobufTypes::MessagePtr RouteConfigProviderManagerImpl::dumpRouteConfigs() {
-  auto config_dump = std::make_unique<envoy::admin::v2::RouteConfigDump>();
-  auto* const dynamic_configs = config_dump->mutable_dynamic_route_configs();
+  auto config_dump = std::make_unique<envoy::admin::v2alpha::RoutesConfigDump>();
+
   for (const auto& provider : getRdsRouteConfigProviders()) {
-    dynamic_configs->Add()->MergeFrom(provider->configAsProto());
+    auto config_info = provider->configInfo();
+    if (config_info) {
+      auto* dynamic_config = config_dump->mutable_dynamic_route_configs()->Add();
```

Comment:
quick c++ question here. In #3129 Matt grabbed the [object by reference](https://github.com/envoyproxy/envoy/pull/3129/files#diff-6ed51949a84ba04d73b5e8a4d1aa768aR184) from the pointer instead of going along with the returned pointer like you are doing.  Can you comment on your preference?

Reply:
It doesn't really matter, what he did is probably more Envoy style. I will change.